### PR TITLE
fix: Clarify comment in backup consumer logic

### PR DESCRIPTION
**Pull request by aiven-anton**

### CODE REVIEW

File: karapace/backup/api.py [hunk 1]
```diff
@@ -298,7 +298,13 @@ def _consume_records(
     if start_offset >= end_offset:
         raise EmptyPartition
 
-    end_offset -= 1  # high watermark to actual end offset
+    # kafka-python returns end offset + 1, i.e. the value that will be assigned
+    # to the next record to be produced. To get the highest offset already
+    # produced we need to subtract 1. Note that this has little to do with high
+    # watermark, which takes into account the highest record offset to be fully
+    # in sync across ISRs, and therefore can be an arbitrary number less than
+    # or equal to the LEO.
```
Comment (eliax1996) [inline karapace/backup/api.py:306]: great, thanks for the explaination! This save a lot of time to the reader of the code. Can we mention also what happens if the topic its empty?

File: karapace/backup/api.py [hunk 2]
```diff
@@ -298,7 +298,13 @@ def _consume_records(
     if start_offset >= end_offset:
         raise EmptyPartition
 
-    end_offset -= 1  # high watermark to actual end offset
+    # kafka-python returns end offset + 1, i.e. the value that will be assigned
```
Comment (matyaskuti) [inline karapace/backup/api.py:301]: kafka-python -> confluent-kafka-python now :)

File: karapace/backup/api.py (hunk 1 above)
Comment (aiven-anton) [reply karapace/backup/api.py]: That would be good, I'll leave that as a room for further future improvement though.

**Comment by aiven-anton** [comment]
Opened this old PR for review, please feel free to merge, and please use squash merge.

**Review by jjaakola-aiven — APPROVED** [review]

**Pull request merged**